File: BackEnd/Application/UseCases/Documents/GetDocuments/GetDocumentsQueryHandler.cs
using Domain.Aggregates;
using Infrastructure.Repository;
using MediatR;
using Microsoft.EntityFrameworkCore;
using SharedKernel.Contants;

namespace Application.UseCases.Documents.GetDocuments;

public class GetDocumentsQueryHandler(IRepository<Document, Guid> documentRepository) : IRequestHandler<GetDocumentsQuery, List<Document>>
{
    public async Task<List<Document>> Handle(GetDocumentsQuery request, CancellationToken cancellationToken)
    {
        // var documents = await documentRepository.GetAll(cancellationToken);
        var documents = await documentRepository.Where(x => x.Status == DocumentStatus.Active).ToListAsync(cancellationToken);
        return documents;
    }
}

File: flutter/lib/application/use_cases/documents/get_documents/get_documents_query_handler.dart
import 'package:enterprise_management/application/use_cases/documents/get_documents/get_documents_query.dart';
import 'package:enterprise_management/infrastructure/data/dtos/documents/document_dto.dart';
import 'package:enterprise_management/infrastructure/repositories/document_repository.dart';
import 'package:enterprise_management/shared_kernel/cqrs/cqrs.dart';
import 'package:enterprise_management/shared_kernel/result/result.dart';

class GetDocumentsQueryHandler extends IQueryHandler<GetDocumentsQuery, List<DocumentDto>> {
  const GetDocumentsQueryHandler({required this._documentRepository});
  final IDocumentRepository _documentRepository;

  @override
  Future<Result<List<DocumentDto>>> handle(GetDocumentsQuery query) async {
    final res = await _documentRepository.getDocuments();
    return Result.success(res);
  }

}